test/Microsoft.AspNetCore.B2CIntegration.Test/AzureAdB2CAuthenticationBuilderExtensionsTests.cs

```diff
@@ -0,0 +1,200 @@
+            services.AddSingleton<ILoggerFactory>(new TestLoggerFactory());
+
+            services.AddAuthentication()
+                .AddAzureAdB2C(o => { })
+                .AddAzureAdB2C(o => { });
+
+            var provider = services.BuildServiceProvider();
+            var b2cOptionsMonitor = provider.GetService<IOptionsMonitor<AzureAdB2COptions>>();
+
+            // Act & Assert
+            var exception = Assert.Throws<InvalidOperationException>(
+                () => b2cOptionsMonitor.Get(AzureAdB2CDefaults.AuthenticationScheme));
+
+            Assert.Equal("An scheme with the name 'AzureADB2C' was already added.", exception.Message);
+        }
+
+        [Fact]
+        public void AddAzureAdB2C_ThrowsWhenOpenIdSchemeIsAlreadyInUse()
+        {
+            // Arrange
+            var services = new ServiceCollection();
+            services.AddSingleton<ILoggerFactory>(new TestLoggerFactory());
+
+            services.AddAuthentication()
+                .AddAzureAdB2C(o => { })
+                .AddAzureAdB2C("B2C", AzureAdB2CDefaults.OpenIdScheme, "Cookie", null, o => { });
+
+            var provider = services.BuildServiceProvider();
+            var b2cOptionsMonitor = provider.GetService<IOptionsMonitor<AzureAdB2COptions>>();
+
+            var expectedMessage = $"The Open ID Connect scheme 'AzureADB2COpenID' can't be associated with the B2C scheme 'B2C'. " +
+                "The Open ID Connect scheme 'AzureADB2COpenID' is already mapped to the B2C scheme 'AzureADB2C'";
+
+            // Act & Assert
+            var exception = Assert.Throws<InvalidOperationException>(
+                () => b2cOptionsMonitor.Get(AzureAdB2CDefaults.AuthenticationScheme));
+
+            Assert.Equal(expectedMessage, exception.Message);
+        }
+
+        [Fact]
+        public void AddAzureAdB2C_ThrowsWhenCookieSchemeIsAlreadyInUse()
+        {
+            // Arrange
+            var services = new ServiceCollection();
+            services.AddSingleton<ILoggerFactory>(new TestLoggerFactory());
+
+            services.AddAuthentication()
+                .AddAzureAdB2C(o => { })
+                .AddAzureAdB2C("B2C", "OpenID", AzureAdB2CDefaults.CookieScheme, null, o => { });
+
+            var provider = services.BuildServiceProvider();
+            var b2cOptionsMonitor = provider.GetService<IOptionsMonitor<AzureAdB2COptions>>();
+
+            var expectedMessage = $"The cookie scheme 'AzureADB2CCookie' can't be associated with the B2C scheme 'B2C'. " +
+                "The cookie scheme 'AzureADB2CCookie' is already mapped to the B2C scheme 'AzureADB2C'";
+
+            // Act & Assert
+            var exception = Assert.Throws<InvalidOperationException>(
+                () => b2cOptionsMonitor.Get(AzureAdB2CDefaults.AuthenticationScheme));
+
+            Assert.Equal(expectedMessage, exception.Message);
+        }
+
+        private class TestLoggerFactory : ILoggerFactory
```

Comment:
We have some built-in "null" implementations of Logger stuff - can you use those? FOr example: https://github.com/aspnet/Logging/blob/dev/src/Microsoft.Extensions.Logging.Abstractions/NullLoggerFactory.cs